sdk/nodejs/appservice/zMixins.ts

```diff
@@ -847,24 +847,28 @@ declare module "./functionApp" {
     }
 }
 
-FunctionApp.prototype.getHostKeys = function(this: FunctionApp) {
+FunctionApp.prototype.getHostKeys = function(this: FunctionApp, attempts?: number) {
     return this.id.apply(async id => {
+        const retryAttempts = attempts || 5
         const credentials = await core.getServiceClientCredentials();
         const client = new AzureServiceClient(credentials);
         const url = `https://management.azure.com${id}/host/default/listkeys?api-version=2018-02-01`;
 
         const response = await client.sendRequest({ method: "POST", url });
-        if (response.status >= 400) {
-            throw new Error(`Failed to retrieve the host keys: ${response.bodyAsText}`);
+        if(response.status < 400) {
+            const body = response.parsedBody;
+            if (body.masterKey === undefined || body.systemKeys === undefined || body.functionKeys === undefined) {
+                throw new Error(`Wrong shape of the host keys response: ${response.bodyAsText}`);
+            }
+            return pulumi.output(body as FunctionHostKeys);
         }
-
-        const body = response.parsedBody;
-        if (body.masterKey === undefined || body.systemKeys === undefined || body.functionKeys === undefined) {
-            throw new Error(`Wrong shape of the host keys response: ${response.bodyAsText}`);
+        if (response.status >= 400 && retryAttempts === 0) {
```

Comment:
If you allow passing attempts as a parameter, it's probably worth changing the check to `<= 0` in case somebody goes crazy and passes a negative number by mistake. I'd prefer not to access that parameter as I mentioned above.